src/components/onboarding/signup/skill-badge.tsx

```diff
@@ -1,28 +1,23 @@
 'use client';
 
-import { Badge } from '../../ui/badge';
 import { X } from 'lucide-react';
-import { Dispatch, SetStateAction } from 'react';
+import { MouseEventHandler } from 'react';
+
+import { Badge } from '../../ui/badge';
 
 type SkillBadgeProps = {
   skill: string;
-  setSkills: Dispatch<SetStateAction<string[]>>;
+  removeSkill: MouseEventHandler<HTMLButtonElement>;
```

Comment:
```suggestion
  onRemove: MouseEventHandler<HTMLButtonElement>;
```
More convential to use a var beginning with `on`. Then functions which handle this callback are called `handleRemove`.

https://react.dev/learn/responding-to-events#naming-event-handler-props